monitoring/nethealth.go

```diff
@@ -142,31 +151,37 @@ func (c *nethealthChecker) check(ctx context.Context, reporter health.Reporter)
 	}
 
 	netData, err := parseMetrics(resp)
-	if trace.IsNotFound(err) {
-		log.Debug("Nethealth counters are uninitialized.")
-		return nil
-	}
 	if err != nil {
 		log.WithError(err).
 			WithField("nethealth-metrics", string(resp)).
 			Error("Received incomplete set of metrics. Could be due to a bug in nethealth or a change in labels.")
 		return nil
 	}
 
-	netData, err = c.filterNetData(netData)
-	if err != nil {
-		log.WithError(err).Error("Failed to filter nethealth data.")
-		return nil
-	}
-
-	updated, err := c.updateStats(netData)
+	updated, err := c.updateStats(filterByK8s(netData, peers))
 	if err != nil {
 		return trace.Wrap(err, "failed to update nethealth stats")
 	}
 
 	return c.verifyNethealth(updated, reporter)
 }
 
+// getPeers returns all nethealth peers as a list of strings.
+func (c *nethealthChecker) getPeers() (peers []string, err error) {
+	opts := metav1.ListOptions{
+		LabelSelector: nethealthLabelSelector.String(),
+		FieldSelector: fields.OneTermNotEqualSelector("metadata.name", c.NodeName).String(),
```

Comment:
I was confused at first - this query was used for nodes, not pods. It will not work for pods as far as I can tell - did you want `spec.nodeName` instead?

Reply:
Ah, yup. This should `spec.nodeName`